### PR TITLE
tests/README.md: Add directory overview

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This adds a small overview of the different subdirectories now in the tests directory

### Testing procedure

- Read the new documentation
  - Check the links
- Check whether I missed any new subdirectory in the list.

### Issues/PRs references

Related to the excellent cleanup work started by @Teufelchen1 and @aabadie

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
 There are a number of tests included in RIOT. They are located in the
-[tests folder](https://github.com/RIOT-OS/RIOT/tree/master/tests). These tests
+[tests directory](https://github.com/RIOT-OS/RIOT/tree/master/tests). These tests
 allow basic functionality to be verified as well as provide an example of
 usage.
 
+# Directory Structure
+
+The [tests directory](https://github.com/RIOT-OS/RIOT/tree/master/tests) in RIOT
+is further divided into a number of directories.
```
Comment (aabadie) [inline tests/README.md:11]:
```suggestion
is further divided into a number of subdirectories.
```
?

Comment (aabadie) [inline tests/README.md:11]:
```suggestion
is further divided into a number of subdirectories.
```
?

Comment (bergzand) [reply tests/README.md]: Thanks, fixed!

**Review by aabadie — APPROVED** [review]
ACK

**Comment by aabadie** [comment]
bors merge

**Pull request merged**